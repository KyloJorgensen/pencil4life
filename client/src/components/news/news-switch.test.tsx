'use strict';

import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import NewsSwitch from './news-switch';

let wrapper: ShallowWrapper;

describe('Render NewsSwitch', () => {
	it('matches the snapshot', () => {
		wrapper = shallow(
			<NewsSwitch/>
		);
		expect(wrapper).toMatchSnapshot();
	});
});