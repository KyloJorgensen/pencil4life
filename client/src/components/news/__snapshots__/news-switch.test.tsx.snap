// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NewsSwitch matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsSwitch />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <NavLink
          activeClassName="selected"
          aria-current="page"
          exact={true}
          to="/news"
        >
          <h2>
            News
          </h2>
        </NavLink>
        <Switch>
          <Route
            component={[Function]}
            path="/news/newitem"
          />
          <Route
            component={[Function]}
          />
        </Switch>
      </div>,
      "className": "news-page-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="selected"
            aria-current="page"
            exact={true}
            to="/news"
          >
            <h2>
              News
            </h2>
          </NavLink>,
          <Switch>
            <Route
              component={[Function]}
              path="/news/newitem"
            />
            <Route
              component={[Function]}
            />
          </Switch>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "selected",
            "aria-current": "page",
            "children": <h2>
              News
            </h2>,
            "exact": true,
            "to": "/news",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "News",
            },
            "ref": null,
            "rendered": "News",
            "type": "h2",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/news/newitem"
              />,
              <Route
                component={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/news/newitem",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <NavLink
            activeClassName="selected"
            aria-current="page"
            exact={true}
            to="/news"
          >
            <h2>
              News
            </h2>
          </NavLink>
          <Switch>
            <Route
              component={[Function]}
              path="/news/newitem"
            />
            <Route
              component={[Function]}
            />
          </Switch>
        </div>,
        "className": "news-page-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="selected"
              aria-current="page"
              exact={true}
              to="/news"
            >
              <h2>
                News
              </h2>
            </NavLink>,
            <Switch>
              <Route
                component={[Function]}
                path="/news/newitem"
              />
              <Route
                component={[Function]}
              />
            </Switch>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "selected",
              "aria-current": "page",
              "children": <h2>
                News
              </h2>,
              "exact": true,
              "to": "/news",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "News",
              },
              "ref": null,
              "rendered": "News",
              "type": "h2",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  path="/news/newitem"
                />,
                <Route
                  component={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/news/newitem",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
